src/Listener/JsonApiListener.php

```diff
@@ -311,6 +311,59 @@ protected function _includeParameter($includes, Subject $subject, $options)
         $this->config('include', $include);
     }
 
+    /**
+     * Parses out fields query parameter and apply it to the query
+     *
+     * @param string|array $fieldSets The query data
+     * @param \Crud\Event\Subject $subject The subject
+     * @param array $options Array of options for includes.
+     * @return void
+     */
+    protected function _fieldSetsParameter($fieldSets, Subject $subject, $options)
+    {
+        if ($fieldSets === null) {
+            return;
+        }
+
+        // format $fieldSets to array acceptable by listener config()
+        $fieldSets = array_map(function ($val) {
+            return explode(',', $val);
+        }, $fieldSets);
+
+        $repository = $subject->query->repository();
+        $associations = $repository->associations();
+
+        $selectFields = [$repository->aliasField($repository->getPrimaryKey())];
+        $columns = $repository->schema()->columns();
+        $contains = [];
+        foreach ($fieldSets as $include => $fields) {
+            if ($include === Inflector::tableize($repository->alias())) {
+                $aliasFields = array_map(function ($val) use ($repository, $columns) {
+                    if (!in_array($val, $columns)) {
+                        return null;
+                    }
+
+                    return $repository->aliasField($val);
+                }, $fields);
+                $selectFields = array_merge($selectFields, array_filter($aliasFields));
```

Comment:
If $aliasFields can be null, shouldnt you check on this before merging?

Reply:
Way to review after a merge :D

Reply:
"merging" the variables, not the pull request :P

Reply:
Since $aliasFields is a result from an `array_map()`, it will always be an array, wouldn't it?